linera-core/src/worker.rs

```diff
@@ -761,85 +824,131 @@ where
         certificate: LiteCertificate<'a>,
         notify_when_messages_are_delivered: Option<oneshot::Sender<()>>,
     ) -> Result<(ChainInfoResponse, NetworkActions), WorkerError> {
-        let full_cert = self.full_certificate(certificate).await?;
-        self.handle_certificate(full_cert, vec![], notify_when_messages_are_delivered)
-            .await
+        match self.full_certificate(certificate).await? {
+            Either::Left(confirmed) => {
+                self.handle_confirmed_certificate(
+                    confirmed,
+                    vec![],
+                    notify_when_messages_are_delivered,
+                )
+                .await
+            }
+            Either::Right(validated) => self.handle_validated_certificate(validated, vec![]).await,
+        }
     }
 
-    /// Processes a certificate.
+    /// Processes a block certificate.
```

Comment:
But it does also process a timeout certificate.
("Block certificate" reads like "validated or confirmed", i.e. one that contains a block?)

Reply:
I'm gonna rm that word from here.